core/src/main/java/pl/netigen/core/ads/AdsManager.kt

```diff
@@ -125,11 +124,24 @@ class AdsManager(var viewModel: NetigenViewModel, val activity: AppCompatActivit
 
     fun isOnline(): Boolean {
         val cm = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        var netInfo: NetworkInfo? = null
-        if (cm != null) {
-            netInfo = cm.activeNetworkInfo
-        }
+        val netInfo: NetworkInfo? = cm.activeNetworkInfo
         return netInfo != null && netInfo.isConnectedOrConnecting
     }
 
+    fun onDestroy() {
+        rewardedAdManager?.onDestroy()
+    }
+
+    fun splashScreenOnStop() {
+        viewModel.isSplashInBackground = true
+    }
+
+    fun splashScreenOnStart() {
+        viewModel.isSplashInBackground = false
+    }
+
+    fun splashScreenOnDestroy() {
+        interstitialAdManager.interstitialAdHandler.removeCallbacksAndMessages(null)
+    }
+
```

Comment:
There is not method for removing rewardAdsListners

Reply:
I didn't find a use for it in Notepad App for I use it only in one place, though there might be some specific cases when you'd like to remove them, for example having rewardedAds in multiple fragments, without removing callbacks it could cause some memory leaks.